Feat: adapt plh_progress_path component for plh_kids_tz application

jfmcquade opened the pull request:
PR Checklist

- [x] PR title descriptive (can be used in release notes)
- [x] merge after #2917 (accidentally included changes)

## Description

- Moves the scope of the dedicated `plh_progress_path` from the `plh_kids_kw` deployment to the general `plh` deployments level
- Exposes new param for `plh_progress_path` component:
  - `starting_postion`
    - Possible values:
      - `"top"`
      - `"bottom"` (default)
    - Default is to start at bottom to avoid breaking changes for existing `plh_kids_kw` deployment
- Adds minor styling overrides for adaptation to `plh_kids_tz` deployment (fine-tuning of step positions in deployment theme overrides)

## Git Issues

Closes #2864

## Screenshots/Videos

[comp_plh_progress_path](https://docs.google.com/spreadsheets/d/130xy0FbCqgVMg8astiR5FBzuUMfVBHtg9k6wmweE1BA/edit?gid=603128459#gid=603128459)

<img width="1200" alt="Screenshot 2025-05-07 at 15 29 13" src="https://github.com/user-attachments/assets/16ca9c3b-f2ee-49d2-afcb-79e4d37d346e" />

<img width="300" alt="Screenshot 2025-05-07 at 15 34 25" src="https://github.com/user-attachments/assets/8d033977-a3b0-41db-b7ee-1a05c838c031" />
<img width="300" alt="Screenshot 2025-05-07 at 15 34 40" src="https://github.com/user-attachments/assets/eaf8e80a-4683-4ba9-898c-e2f982e45fa8" />

esmeetewinkel reviewed (approved):
Thanks! Attempted to integrate in `plh_kids_tz` deployment, content PR [here](https://github.com/ParentingForLifelongHealth/plh-kids-app-tz-content/pull/184). 

Two minor comments: 
1. On all screen widths I'm getting a bit of padding between the map component and the header. Do you have a suggestion of how to remove it? I've been playing around with negative padding but wasn't particularly successful (the amount of negative padding needed seemed to depend on the screen width?). 

2. On narrow screens, there is a bit of space between the first module list item and the horizon. Could it drop slightly so that it always overlaps with the entire start of the path? 

<img width="327" alt="image" src="https://github.com/user-attachments/assets/dab601e5-9aa1-4f17-ad94-d53f155425e4" />

jfmcquade commented:
Thanks @esmeetewinkel, I've addressed your points:

> 1. On all screen widths I'm getting a bit of padding between the map component and the header. Do you have a suggestion of how to remove it? I've been playing around with negative padding but wasn't particularly successful (the amount of negative padding needed seemed to depend on the screen width?).

This is due to a `margin-top` that gets applied to all template rows by default, so the way to override it is to set `margin-top: 0;` on the component template row, e.g.:

<img width="800" alt="Screenshot 2025-05-08 at 12 08 30" src="https://github.com/user-attachments/assets/be6b6caf-4b1b-42eb-beae-3759faae9d84" />

> 2. On narrow screens, there is a bit of space between the first module list item and the horizon. Could it drop slightly so that it always overlaps with the entire start of the path?

I've tweaked the spacing (with https://github.com/IDEMSInternational/open-app-builder/pull/2885/commits/41d06d848f5d4c8764479a0b7ffd77acee807b64), so that now the item comfortably overlaps the horizon on both large and small screens:

<img width="286" alt="Screenshot 2025-05-08 at 12 06 43" src="https://github.com/user-attachments/assets/cc41b869-c56a-4c12-8ee0-efda9197ae1c" />

<img width="365" alt="Screenshot 2025-05-08 at 12 06 32" src="https://github.com/user-attachments/assets/e5c94312-18fb-4111-bd01-c071cd379e1b" />

esmeetewinkel reviewed (approved):
Thanks @jfmcquade, that works! 

![image](https://github.com/user-attachments/assets/3d39df55-cbba-439d-bb8d-33b8f7052a95)
